index.js

```diff
@@ -1,23 +1,102 @@
+const cluster = require('cluster');
+const twilio = require('twilio');
+require('dotenv').config();
+
 const Chatbot = require('./src/Chatbot');
 const Rivebot = require('./src/Rivebot');
 const Updater = require('./src/Updater');
 const Messenger = require('./src/Messenger');
 const constants = require('./src/constants');
-require('dotenv').config();
 const api = require('./src/api');
-const Botkit = require('botkit');
 const server = require('./server.js');
 const helpers = require('./helpers');
 
-// Create the Botkit controller, which controls all instances of the bot.
-const twilioController = Botkit.twiliosmsbot({
-  account_sid: process.env.TWILIO_ACCOUNT_SID,
-  auth_token: process.env.TWILIO_AUTH_TOKEN,
-  twilio_number: process.env.TWILIO_NUMBER
-});
+const numCPUs = require('os').cpus().length;
+let workerCount = 0;
+
+/**
+ * This creates node cluster workers for bot app
+ * Workers maximizes the CPU utility and ensure that the bot runs with less likelihood of a downtime
+ * For more info, see: https://www.sitepoint.com/how-to-create-a-node-js-cluster-for-speeding-up-your-apps/
+ * and https://medium.com/the-andela-way/scaling-out-with-node-clusters-1dca4a39a2a
+ */
+if (cluster.isMaster) {
+  console.log(`Master ${process.pid} is running`);
+
+  //Creates as many workers as the number of available CPUs
+  while (workerCount <= numCPUs) {
+    cluster.fork();
+    workerCount += 1;
+  }
+
+  // Starts a new worker when one worker exits/dies
+  cluster.on('exit', (worker, code, signal) => {
+    console.log(
+      'worker %d died (%s). restarting...',
+      worker.process.pid, signal || code
+    );
+    cluster.fork();
+  });
+} else {
+  // Set up an Express-powered webserver to webhook endpoints
+  server(
+    fbEndpoint,
+    twilioReceiveSmsController,
+    getCoachResponse,
+    testTwilioCredentialsController
+  );
+  console.log(
+    `Worker ${process.pid} started
+    listening on server port ${process.env.PORT || 3002}`
+  );
+}
+
+/**
+ * This function is responsible for receiving the SMS from Twilio
+ * @param {object} request 
+ * @param {object} response 
+ */
+async function twilioReceiveSmsController (request, response) {
+  const message = request.body;
+  const userPlatformId = message.From;
+  const userMessage = message.Body;
+  await run({
+    platform: constants.SMS,
+    userPlatformId,
+    userMessage
+  });
+}
+
+async function testTwilioCredentialsController (request, response) {
```

Comment:
Does this method really need to be in the codebase?

Reply:
Yes.
A better alternative would have been implementing Twilio in the coach app and to use the twilio SMS API there. When I tried it, some dependencies in the Twilio npm package causes the coach app server to fail on `npm start`.

This was the only option I could think of.